tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+import dataclasses
+import uuid
+from string import Template
+from typing import Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.steps.provision.testcloud import GuestTestcloud
+from tmt.utils import field
+
+DEFAULT_IMAGE_BUILDER = "quay.io/centos-bootc/bootc-image-builder:latest"
+
+
+class GuestBootc(GuestTestcloud):
+    containerimage: str
+
+    def __init__(self,
+                 *,
+                 data: tmt.steps.provision.GuestData,
+                 name: Optional[str] = None,
+                 parent: Optional[tmt.utils.Common] = None,
+                 logger: tmt.log.Logger,
+                 containerimage: Optional[str]) -> None:
```

Comment:
ah, yea that’s right. I had originally marked containerimage as optional because I had set `_guest_class = GuestBootc` which caused the `GuestBootc` class to be instantiated without a `containerimage`. Using `GuestBootc` as the `_guest_class` was causing some other odd issues (the tests would pass when running them individually, but when running multiple tests only the first would pass) so I flipped it back to GuestTestcloud which seems to work as expected.

Reply:
I'm still convinced this can be improved: given the code in `go()`, `containerimage` will never be `None`. It will either be a string returned by `_build_derived_image()`, or a string returned by `_build_base_image()`, in any case `containerimage` variable in `go()` will always contain a valid string. Therefore this parameter does not need to be `Optional`, and the test below is also superfluous.